go/cmd/dolt/commands/merge.go

```diff
@@ -93,146 +99,274 @@ func (cmd MergeCmd) Exec(ctx context.Context, commandStr string, args []string,
 		defer closeFunc()
 	}
 
-	if apr.ContainsAll(cli.SquashParam, cli.NoFFParam) {
-		cli.PrintErrf("error: Flags '--%s' and '--%s' cannot be used together.\n", cli.SquashParam, cli.NoFFParam)
+	ok := validateDoltMergeArgs(apr, usage, cliCtx)
+	if ok != 0 {
 		return 1
 	}
 
-	// This command may create a commit, so we need user identity
-	if !cli.CheckUserNameAndEmail(cliCtx.Config()) {
+	// allows merges that create conflicts to stick
+	_, _, err = queryist.Query(sqlCtx, "set @@dolt_force_transaction_commit = 1")
+	if err != nil {
+		cli.Println(err.Error())
 		return 1
 	}
 
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), help)
+	query, err := constructInterpolatedDoltMergeQuery(apr)
+	if err != nil {
+		cli.Println(err.Error())
+		return 1
+	}
+	_, _, err = queryist.Query(sqlCtx, query)
+	if err != nil {
+		cli.Println(err.Error())
+		return 1
+	}
+	_, _, err = queryist.Query(sqlCtx, "COMMIT")
+	if err != nil {
+		cli.Println(err.Error())
+		return 1
 	}
 
-	var verr errhand.VerboseError
-	if apr.Contains(cli.AbortParam) {
-		mergeActive, err := isMergeActive(ctx, dEnv)
+	_, _, err = queryist.Query(sqlCtx, "set @@dolt_force_transaction_commit = 0")
+	if err != nil {
+		cli.Println(err.Error())
+		return 1
+	}
+
+	// calculate merge stats
+	if !apr.Contains(cli.AbortParam) && !apr.Contains(cli.NoCommitFlag) {
```

Comment:
Everything which fails from here on down should be scrutinized. Remember that the state of the DB has changed, and someone could be depending on the exit status of this program to control some other business logic. I wonder if failure to print statistics merits an exit status of 1. Thoughts?

Reply:
All of the stats logic should be in it's own function. I'd like to split out printing here, but that seems kind of tangled in with this.